ysh/cpython.py

```diff
@@ -70,6 +70,15 @@ def _PyObjToValue(val):
         #    return value.BashArray(shell_array)
         return value.List(typed_array)
 
+    elif isinstance(val, xrange):
+        # awkward, but should go away once everything is typed...
+        l = list(val)
+        if len(l) > 1:
+            return value.Range(l[0], l[-1])
+
+        # Empty range
+        return value.Range(0, 0)
```

Comment:
Added this only because of a compulsion for symmetry. Let me check that it's actually necessary...

Reply:
Looks like the spec tests don't depend on this, but we will need this for functions that return ranges